lib/core-integration/src/Test/Integration/Scenario/API/Shelley/StakePools.hs

```diff
@@ -552,6 +555,137 @@ spec = describe "SHELLEY_STAKE_POOLS" $ do
+            nonRetiredPoolIds <- eventually "One of the pools should retire." $ do
+                response <- listPools ctx arbitraryStake
+                verify response [ expectListSize 3 ]
+                getFromResponse Prelude.id response
+                    & fmap (view (#id . #getApiT))
+                    & Set.fromList
+                    & pure
+            let reportError = error $ unlines
+                    [ "Unable to find a retired pool ID."
+                    , "Test cluster pools:"
+                    , unlines (showT <$> Set.toList testClusterPoolIds)
+                    , "Non-retired pools:"
+                    , unlines (showT <$> Set.toList nonRetiredPoolIds)
+                    ]
+            let retiredPoolIds =
+                    testClusterPoolIds `Set.difference` nonRetiredPoolIds
+            let retiredPoolId =
+                    fromMaybe reportError $ listToMaybe $ Set.toList retiredPoolIds
+            w <- fixtureWallet ctx
+            r <- joinStakePoolUnsigned @n @'Shelley ctx w (ApiT retiredPoolId)
+            expectResponseCode HTTP.status404 r
+            expectErrorMessage (errMsg404NoSuchPool (toText retiredPoolId)) r
+
+    describe "STAKE_POOLS_JOIN_UNSIGNED_04"
+        $ it "Cannot join a pool that's never existed" $ \ctx -> do
+            (Right non_existing_pool_id) <- pure $ decodePoolIdBech32
+                "pool1y25deq9kldy9y9gfvrpw8zt05zsrfx84zjhugaxrx9ftvwdpua2"
+            w <- fixtureWallet ctx
+            r <- joinStakePoolUnsigned @n @'Shelley ctx w (ApiT non_existing_pool_id)
+            expectResponseCode HTTP.status404 r
+            expectErrorMessage (errMsg404NoSuchPool (toText non_existing_pool_id)) r
+
+    describe "STAKE_POOLS_QUIT_UNSIGNED_01"
+        $ it "Can quit a joined pool" $ \ctx -> do
+            w <- fixtureWallet ctx
+
+            pool:_ <- map (view #id) . snd <$> unsafeRequest @[ApiStakePool]
+                ctx (Link.listStakePools arbitraryStake) Empty
+
+            joinStakePool @n ctx pool (w, fixturePassphrase) >>= flip verify
+                [ expectResponseCode HTTP.status202
+                , expectField (#status . #getApiT) (`shouldBe` Pending)
+                , expectField (#direction . #getApiT) (`shouldBe` Outgoing)
+                ]
+
+            eventually "Wallet is delegating to p1" $ do
+                request @ApiWallet ctx (Link.getWallet @'Shelley w)
+                    Default Empty >>= flip verify
+                    [ expectField #delegation (`shouldBe` delegating pool [])
+                    ]
+
+            -- Quit Pool
+            quitStakePoolUnsigned @n @'Shelley ctx w >>= \o -> do
+                verify o
+                    [ expectResponseCode HTTP.status200
+                    , expectField #inputs (`shouldSatisfy` (not . null))
+                    , expectField #outputs (`shouldSatisfy` (not . null))
+                    , expectField #certificates (`shouldSatisfy` (not . null))
```

Comment:
Same here, we could assert that the certificate is indeed of type "quit_pool" (and that there's no other certificates present)